java/piranha/src/test/java/com/uber/piranha/EnumConstantTest.java

```diff
@@ -848,4 +848,236 @@ public void testRemoveEnumWithNonEnumConstantConstructor() {
+            "import java.lang.annotation.Retention;",
+            "import java.lang.annotation.RetentionPolicy;",
+            "import java.lang.annotation.Target;",
+            "class TestEnumInAnnotation {",
+            " enum TestExperimentName {",
+            "  OTHER_FLAG,",
+            "  STALE_FLAG;",
+            " }",
+            " @Retention(RetentionPolicy.RUNTIME)",
+            " @Target({ElementType.METHOD})",
+            " @interface ToggleTesting {",
+            "  TestExperimentName[] treated();",
+            " }",
+            " @ToggleTesting(treated = TestExperimentName.STALE_FLAG)",
+            " public void annotation_test() {}",
+            "}")
+        .addOutputLines(
+            "TestEnumInAnnotation.java",
+            "package com.uber.piranha;",
+            "import java.lang.annotation.ElementType;",
+            "import java.lang.annotation.Retention;",
+            "import java.lang.annotation.RetentionPolicy;",
+            "import java.lang.annotation.Target;",
+            "class TestEnumInAnnotation {",
+            " enum TestExperimentName {",
+            "  OTHER_FLAG;",
+            " }",
+            " @Retention(RetentionPolicy.RUNTIME)",
+            " @Target({ElementType.METHOD})",
+            " @interface ToggleTesting {",
+            "  TestExperimentName[] treated();",
+            " }",
+            " public void annotation_test() {}",
+            "}")
+        .doTest();
+  }
+
+  /**
+   * Test to cleanup the first enum in annotation. Should remove the enum and the comma after it and
+   * keep the annotation and the other enum
+   */
+  @Test
+  public void testRemoveFirstEnumFromAnnotationRemovingCommaAndKeepingOtherEnum() {
+    ErrorProneFlags.Builder b = ErrorProneFlags.builder();
+    b.putFlag("Piranha:FlagName", "STALE_FLAG");
+    b.putFlag("Piranha:IsTreated", "true");
+    b.putFlag("Piranha:Config", "config/properties.json");
+
+    BugCheckerRefactoringTestHelper bcr =
+        BugCheckerRefactoringTestHelper.newInstance(new XPFlagCleaner(b.build()), getClass());
+
+    bcr = bcr.setArgs("-d", temporaryFolder.getRoot().getAbsolutePath());
+
+    bcr.addInputLines(
+            "TestEnumInAnnotation.java",
+            "package com.uber.piranha;",
+            "import java.lang.annotation.ElementType;",
+            "import java.lang.annotation.Retention;",
+            "import java.lang.annotation.RetentionPolicy;",
+            "import java.lang.annotation.Target;",
+            "class TestEnumInAnnotation {",
+            " enum TestExperimentName {",
+            "  OTHER_FLAG,",
+            "  STALE_FLAG;",
+            " }",
+            " @Retention(RetentionPolicy.RUNTIME)",
+            " @Target({ElementType.METHOD})",
+            " @interface ToggleTesting {",
+            "  TestExperimentName[] treated();",
+            " }",
+            " @ToggleTesting(treated = {TestExperimentName.STALE_FLAG, TestExperimentName.OTHER_FLAG})",
```

Comment:
Another test could be added `@ToggleTesting(treated = {TestExperimentName.STALE_FLAG,TestExperimentName.OTHER_FLAG})` and 
```
@ToggleTesting(treated = {TestExperimentName.STALE_FLAG,
                                                TestExperimentName.OTHER_FLAG})
```